cmd/apiserver-watcher/azure_handler.go

```diff
@@ -0,0 +1,167 @@
+package main
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/coreos/go-iptables/iptables"
+	"github.com/golang/glog"
+	"github.com/vishvananda/netlink"
+)
+
+// Prevent hairpin traffic when the apiserver is up
+
+// As per the Azure documentation (https://docs.microsoft.com/en-us/azure/load-balancer/concepts//limitations),
+// if a backend is load-balanced to itself, then the traffic will be dropped.
+//
+// This is because the L3LB does DNAT, so while the outgoing packet has a destination
+// IP of the VIP, the incoming load-balanced packet has a destination IP of the
+// host. That means that it "sees" a syn with the source and destination
+// IPs of itself, and duly replies wit a syn-ack back to itself. However, the client
+// socket expects a syn-ack with a source IP of the VIP, so it drops the packet.
+//
+// The solution is to redirect traffic destined to the lb vip back to ourselves.
+//
+// Example rules using vip 10.0.0.1:
+//
+// -A PREROUTING -m comment --comment "azure LB vip overriding for pods" -j azure-vips
+// -A OUTPUT -m comment --comment "azure LB vip overriding for local clients" -j azure-vips
+// -A azure-vips --dst 10.0.0.1 -j REDIRECT
+
+type azureHandler struct {
+	vip    string
+	ipt    *iptables.IPTables
```

Comment:
woud love this to be iptable or iptbl

Reply:
`ipt` is the way it is implemented in kube-proxy, I was keeping the same naming

Reply:
cool fair point

Reply:
naming is hard :upside_down_face: